tools/sof-dump-status.py

```diff
@@ -82,6 +92,31 @@ def loadDMI(self):
             self.dmi[name] = line[idx:]
         pass
 
+    def loadDT(self):
+        def _loadDT(filename):
+            # path to file containing SoC-related information
+            filepath = "/sys/devices/soc0/" + filename
+
+            # since the name of the files found at info_path perfectly match
+            # the names of the SoC attributes, they can be used as keys in
+            # dt_info dictionary
+            self.dt_info[filename] = read_file(filepath).strip()
+
+        try:
+            # load SoC id
+            _loadDT("soc_id")
+            # load SoC family
+            _loadDT("family")
+            # load SoC revision
+            _loadDT("revision")
+            # load machine name
+            _loadDT("machine")
+        except IOError:
+            # if any of the above calls fails, then we're probably
+            # not on a DT-based platform so there should be no
+            # information in dt_info
```

Comment:
This looks a bit weird. Is there no concept of required versus optional attributes?

Reply:
In the current implementation there isn't, but I think it would be a good idea to add said concepts. Also, I think I'm going to remove

```
# load SoC family
_loadDT("family")
# load SoC revision
 _loadDT("revision")
# load machine name
_loadDT("machine")
```

altogether since they aren't really used for anything at the moment. The only reason I left those in was to allow `dumpPlatform` to print information about the SoC in the future, but I think it would be a much better idea to just leave those out for now, add the concept of optional/required attributes and mark them as optional if they'll be required in the future.